caluma/caluma_workflow/events.py

```diff
@@ -1,24 +1,66 @@
+import sys
+
 from django.dispatch import Signal
 
+from caluma.caluma_core.events import send_event
+
+module = sys.modules[__name__]
+
+MODEL_ACTIONS = {
+    "work_item": ["create", "complete", "cancel", "skip", "suspend", "resume"],
+    "case": ["create", "complete", "cancel", "suspend", "resume"],
+}
+
+# create "pre|post"_"create|complete|..."_"work_item|case" events in all given combinations
+for obj, actions in MODEL_ACTIONS.items():
+    for action in actions:
+        for prefix in ["pre", "post"]:
+            setattr(
+                module,
+                f"{prefix}_{action}_{obj}",
+                Signal(providing_args=[obj, "user", "context"]),
+            )
+
+# TODO: remove in the next major release since those are events are deprecated...
+# ... in favor of "post_" events
 created_work_item = Signal(providing_args=["work_item", "user", "context"])
-canceled_work_item = Signal(providing_args=["work_item", "user", "context"])
 completed_work_item = Signal(providing_args=["work_item", "user", "context"])
 skipped_work_item = Signal(providing_args=["work_item", "user", "context"])
 suspended_work_item = Signal(providing_args=["work_item", "user", "context"])
 resumed_work_item = Signal(providing_args=["work_item", "user", "context"])
 completed_case = Signal(providing_args=["case", "user", "context"])
 created_case = Signal(providing_args=["case", "user", "context"])
-canceled_case = Signal(providing_args=["case", "user", "context"])
 suspended_case = Signal(providing_args=["case", "user", "context"])
 resumed_case = Signal(providing_args=["case", "user", "context"])
 
-# TODO: remove in the next major release since those are events are deprecated
-# in favor of the american notation
+# ... also in favor of the american notation
 cancelled_work_item = Signal(providing_args=["work_item", "user", "context"])
 cancelled_case = Signal(providing_args=["case", "user", "context"])
 
-for (event, old, new) in [
-    (cancelled_case, "cancelled_case", "canceled_case"),
-    (cancelled_work_item, "cancelled_work_item", "canceled_work_item"),
-]:
-    event._deprecation_reason = f"`caluma_workflow.events.{old}` is deprecated in favor of `caluma_workflow.events.{new}` and will be removed in the next major version."
+DEPRECATIONS = {
+    "post_create_work_item": "created_work_item",
+    "post_complete_work_item": "completed_work_item",
+    "post_cancel_work_item": "cancelled_work_item",
```

Comment:
This event has multiple deprecations:

- cancelled_work_item
- canceled_work_item